src/main/java/edu/ie3/datamodel/utils/validation/ValidationUtils.java

```diff
@@ -125,27 +183,79 @@ else if (ThermalUnitInput.class.isAssignableFrom(assetInput.getClass()))
+      return List.of(
+          new Failure<>(
+              new InvalidEntityException(
+                  "Validation not possible because received object {"
+                      + assetTypeInput
+                      + "} was null",
+                  e)));
+    }
+
+    List<Try<Void, ? extends ValidationException>> exceptions = new ArrayList<>();
+
     if (assetTypeInput.getUuid() == null)
-      throw new InvalidEntityException("No UUID assigned", assetTypeInput);
+      exceptions.add(new Failure<>(new InvalidEntityException("No UUID assigned", assetTypeInput)));
     if (assetTypeInput.getId() == null)
-      throw new InvalidEntityException("No ID assigned", assetTypeInput);
+      exceptions.add(new Failure<>(new InvalidEntityException("No ID assigned", assetTypeInput)));
 
     // Further checks for subclasses
     if (LineTypeInput.class.isAssignableFrom(assetTypeInput.getClass()))
-      ConnectorValidationUtils.checkLineType((LineTypeInput) assetTypeInput);
+      exceptions.addAll(ConnectorValidationUtils.checkLineType((LineTypeInput) assetTypeInput));
     else if (Transformer2WTypeInput.class.isAssignableFrom(assetTypeInput.getClass()))
-      ConnectorValidationUtils.checkTransformer2WType((Transformer2WTypeInput) assetTypeInput);
+      exceptions.addAll(
+          ConnectorValidationUtils.checkTransformer2WType((Transformer2WTypeInput) assetTypeInput));
     else if (Transformer3WTypeInput.class.isAssignableFrom(assetTypeInput.getClass()))
-      ConnectorValidationUtils.checkTransformer3WType((Transformer3WTypeInput) assetTypeInput);
+      exceptions.addAll(
+          ConnectorValidationUtils.checkTransformer3WType((Transformer3WTypeInput) assetTypeInput));
     else if (SystemParticipantTypeInput.class.isAssignableFrom(assetTypeInput.getClass()))
-      SystemParticipantValidationUtils.checkType((SystemParticipantTypeInput) assetTypeInput);
+      exceptions.addAll(
+          SystemParticipantValidationUtils.checkType((SystemParticipantTypeInput) assetTypeInput));
     else {
-      throw checkNotImplementedException(assetTypeInput);
+      exceptions.add(
+          new Failure<>(
+              new FailedValidationException(
+                  checkNotImplementedException(assetTypeInput).getMessage())));
     }
+
+    return exceptions;
+  }
+
+  /**
+   * Checks the validity of the ids for a given set of {@link AssetInput}.
+   *
+   * @param inputs a set of asset inputs
+   * @return a list of try objects either containing an {@link UnsafeEntityException} or an empty
+   *     Success
+   */
+  protected static List<Try<Void, UnsafeEntityException>> checkIds(
+      Set<? extends AssetInput> inputs) {
+    List<String> ids = new ArrayList<>();
+    List<Try<Void, UnsafeEntityException>> exceptions = new ArrayList<>();
+
+    inputs.forEach(
+        input -> {
+          String id = input.getId();
+          if (!ids.contains(id)) {
+            ids.add(id);
+            exceptions.add(Success.empty());
```

Comment:
Since we're not really interested in the Void-successes produced during validation, maybe we can skip adding these here?